tools/plugin-base/src/main/kotlin/io/spine/tools/gradle/UsefulSetProperty.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.gradle
+
+import com.google.common.collect.ImmutableSet
+import kotlin.reflect.KClass
+import org.gradle.api.Project
+import org.gradle.api.provider.SetProperty
+
+/**
+ * A Gradle `SetProperty` which has useful utility methods.
+ */
+public class UsefulSetProperty<T : Any>
+
+/**
+ * Creates a new `UsefulSetProperty`.
+ *
+ * @param project
+ *          the project which provides an object factory for creating Gradle properties
+ * @param klass
+ *          type of the elements of the set
+ */
+constructor(
+    project: Project, klass: KClass<T>
+) : SetProperty<T> by project.objects.setProperty(klass.java) {
+
+    /**
+     * Creates a new `UsefulSetProperty`.
+     *
+     * This is a Java-friendly constructor. Use the primary constructor in Kotlin.
+     *
+     * @param project
+     *          the project which provides an object factory for creating Gradle properties
+     * @param cls
+     *          Java class of the elements of the set
+     */
+    public constructor(project: Project, cls: Class<T>) : this(project, cls.kotlin)
+
+    /**
+     * Obtains the value of this property and applies the given mapping function to each element.
+     */
+    public fun <O> transform(mapper: (T) -> O): ImmutableSet<O> =
```

Comment:
Please think of a different letter for the generic parameter. `<0>` looks  a bit misleading. Please also document the generic parameter.